spring-integration-core/src/main/java/org/springframework/integration/handler/support/MessagingMethodInvokerHelper.java

```diff
@@ -687,6 +689,20 @@ private boolean contentTypeIsJson(Message<?> message) {
 		final AtomicReference<Class<?>> ambiguousFallbackType = new AtomicReference<>();
 		final AtomicReference<Class<?>> ambiguousFallbackMessageGenericType = new AtomicReference<>();
 		final Class<?> targetClass = getTargetClass(targetObject);
+
+		final String methodName;
+
+		if (methodNameToUse == null) {
```

Comment:
It's not good practice to mutate parameter arguments. Better to leave the argument final (e.g. `final String methodNameArg`) and then `String methodName = methodNameArg`.